Qiskit/qiskit-ibm-transpiler
# Refactor tests

### Details and comments
**Done in this PR:**
- Unify all the synthesis passes tests (different types and local+cloud mode) on one file. Since local synthesis for Pauli doesn't exist yet, there is a `skip` for that case
- Unify all the collector passes tests (different types and local+cloud mode) on one file
- Unify local and cloud routing pass tests on one file
- Rename passes files for easy reading
- Group all tests on the same folder
- Unify conftest files
- Group all the fixtures on the conftest file
- Create parametrize functions to not repeat parametrize configs
- Create a file for parametrize functions that are common to more than one file.
- Add several TODO and FIXME comments for things that I think we should review
- Fix some errors on the code found when refactoring the tests

**More posible refactor where I would like to hear your thoughts:**
- We could move all the parametrize functions to the same file the same way we have with fixtures, so we can see easily what is done and if we can reuse it. Also, we could even create structures with their values so, in the case of adding a new tests we just had to change one place. Example: most of the parametrizations has the Collector, so we could create a dict with that info and access to the dict on the parametrize function.
- We could create a new file for utils. Right now we have 2 functions defined to create circuits for the tests  (plus other functions imported). We could create that file to have that creations together. Although I think we could first review the functions just in case we are creating several functions for doing the same
- Some TranspilerService tests are similar to other passes tests like: wrong token, wrong url, exceed time, .... Should I remove that tests from the passes?

Thanks!
